src/cascadia/TerminalControl/ControlCore.idl

```diff
@@ -29,6 +29,16 @@ namespace Microsoft.Terminal.Control
         All
     };
 
+    struct SelectionMarkerMetadata

```

Comment:
Is this specifically "Marker metadata" or is it "Selection info"? It feels like it could be the second, with some gentle massaging.

Does the **CORE** need to be in charge of telling us when to flip it? Can the Control not figure that out on its own?

Reply:
Fair enough. Made everything more generic. 
- `SelectionMarkerMetadata SelectionMarkerInfo` --> `SelectionData SelectionInfo`
- `FlipStartMarker` --> `StartAtLeftBoundary`
- `FlipEndMarker` --> `EndAtRightBoundary`

The `flip` ones are annoying because I don't think `TermControl` should know anything about the buffer's size. But for the purpose of rendering the marker, it makes sense to say "start is at the left boundary", I feel.

```cpp
// ControlCore::SelectionInfo()
info.StartAtLeftBoundary = _terminal->GetSelectionAnchor().x == bufferSize.Left();
info.EndAtRightBoundary = _terminal->GetSelectionEnd().x == bufferSize.RightInclusive();
```

Reply:
We poked "bufferHeight" through the Core winrt interface, and TermControl uses it... so I dunno! The boundary names seem fine, but it still seems like a hack. We can probably ship with it, especially if our plan is to move it into the renderer later.